### DISCUSSION TITLE
Livewire search with pagination

### DISCUSSION THREAD

**Discussion by Yusufjonov707** [livewire/livewire#8312] [1. Help]
public function updatingPage()
    {
        $this->setPage(1, pageName: 'roads-page');
    }
    #[Computed]
    public function roads()
    {
        // $name = 'roads-page';
        $roads = Roads::query();

        $roads->where(function($query) {
            if ($this->search !== '') { 
                $query->where('name_uz', 'like', '%' . $this->search . '%');
            }
            
            if ($this->znach_uz !== '') { 
                $query->where('znach_uz', 'like', '%' . $this->znach_uz . '%');
            }
        });
      
        $roads = $roads->paginate(20, pageName: 'roads-page');
        return $roads;
    }
    do this has to be like this ?

**Comment by mehdit79** [comment]
I think you should set query in render function of class

**Comment by Yusufjonov707** [comment]
and am I have to call function in render ?

**Reply by mehdit79** [reply]
yes, actually

**Comment by joshhanley** [comment]
@Yusufjonov707 actually a computed property like you have is cleaner than putting it in the render method and it's how I would recommend doing it.

But to call a computed property in your view, you need to call `$this->roads` instead of just `$roads` and then it should work.

**Reply by Yusufjonov707** [reply]
thanks for helping it working fine now . 
public function updatingPage()
{
    $this->setPage(1, pageName: 'roads-page');
}
I put everything inside render and called this->updatingPage() function before render returns and setPage() started working.
